dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/layers/LayerBase.java

```diff
@@ -39,13 +46,18 @@ public final int getIndex() {
   }
 
   /**
-   * @return the number of layer output nodes.
+   * @return the shape of inputs.
    */
-  public final int getNumOutput() {
-    return this.numOutput;
+  public final int[] getInputShape() {
```

Comment:
Generating one-hot output vectors was the main reason why `getNumOutput()` exists. However, `getOutputShape()` is needed to provide information about the shape of inputs to subsequent layers at `getLayerInstances()` in `NeuralNetworkUtils`. Actually, `getInputShape()` does not have to be public. But, it may offer more information about layers.